icu4c/source/i18n/calendar.cpp

```diff
@@ -1236,16 +1236,33 @@ Calendar::set(int32_t year, int32_t month, int32_t date, int32_t hour, int32_t m
 // -------------------------------------
 int32_t Calendar::getRelatedYear(UErrorCode &status) const
 {
-    return get(UCAL_EXTENDED_YEAR, status);
+    int32_t year = get(UCAL_EXTENDED_YEAR, status);
+    if (U_FAILURE(status)) {
+        return 0;
+    }
+    int32_t differences = getRelatedYearDifferences();
+    if (differences != 0 && uprv_add32_overflow(year,  differences, &year)) {
```

Comment:
Stylistic nit: `uprv_add32_overflow()` actually does stuff, but it's buried inside an `if`, and moreover, it's the second of two conditions in the `if`.  It's not a big deal in a short function like this one, but I feel like that kind of thing tends to obscure the logic when you're reading the code.